main.go

```diff
@@ -97,15 +100,37 @@ func runMain() error {
 	if err := notify.Watch(conf.Directory, fsEvents, notify.InCloseWrite, notify.InMovedTo); err != nil {
 		return err
 	}
+
+	nextFailedRetry := time.Now()
+	failedRetryAttempts := 0
+	var success bool
 	defer notify.Stop(fsEvents)
 	for {
 		// Check for files to upload first in case there are CPTV
 		// files around when the uploader starts.
 		cr.Start()
 		cr.WaitUntilUpLoop(connectionTimeout, connectionRetryInterval, -1)
-		if err := uploadFiles(apiClient, conf.Directory); err != nil {
+		if success, err = uploadFiles(apiClient, conf.Directory); err != nil {
 			return err
 		}
+
+		//try failed uploads again if succeeded
+		if success && time.Now().After(nextFailedRetry) {
+			if retryFailedUploads(apiClient, conf.Directory) {
+				failedRetryAttempts = 0
+				nextFailedRetry = time.Now()
+			} else {
+				failedRetryAttempts += 1
+				timeAddition := time.Duration(failedRetryInterval.Seconds()*float64(failedRetryAttempts*failedRetryAttempts)) * time.Second
```

Comment:
Nice backoff approach. I think the code can be simpler though. This should be identical:

```go
timeAddition := failedRetryInterval * time.Duration(failedRetryAttempts*failedRetryAttempts)
``` 

The thing to remember is that a time.Duration is just a uint64 (nanoseconds since the Unix epoch time) with a bunch of methods added.